src/full/Agda/TypeChecking/Serialise/Instances/Common.hs

```diff
@@ -434,15 +425,17 @@ instance EmbPrj a => EmbPrj (Arg a) where
 
   value = valueN Arg
 
-instance EmbPrj HasEta where
-  icod_ YesEta = icodeN' YesEta
-  icod_ NoEta  = icodeN 1 NoEta
+instance EmbPrj a => EmbPrj (HasEta' a) where
+  icod_ YesEta    = icodeN' YesEta
+  icod_ (NoEta a) = icodeN 1 NoEta a
```

Comment:
I guess we don't need the tag: `[]` is `YesEta` and `[a]` can only come from `NoEta`

Reply:
Ok, I will fix this.